general/README.md

```diff
@@ -208,15 +209,102 @@ Avoid variable indirection when possible. Reassigning variables when no transfor
 ```js
 // good
 _handleEvent = function(e) {
-    _doSomethingWithEvent(e.target.checked);   
+    _doSomethingWithEvent(e.target.checked);
 }
 
 // bad
 _handleEvent = function(e) {
     var checked = e.target.checked;
 
-    _doSomethingWithEvent(checked);   
+    _doSomethingWithEvent(checked);
 }
 ```
 
 **[⬆ back to top](#table-of-contents)**
+
+## Naming Conventions
+> There are only two hard things in Computer Science: cache invalidation and naming things. *Phil Karlton*
+
+### Variables: Classes or Newable Functions (meant to be factories).
+It should be singular and begin with a capital letter.
+
+```js
+var ticket = new Ticket(id: 555);
+```
+### Variables: helpers and commmon variables
+
+It should always start with a lower case and be camel Case if is composed by 2 words.
+
+```js
+    var pagination = require('pagination');
+
+    pages = pagination.getPages();
+```
+
+### Varialbles: boolean
+
+It sholud start either with `is` or `has`
+
+```js
+var isAvailable = true,
+    hasTickets = false;
+```
+### Variables: jquery element or elements
+
+It sholud start with a dollar sign (`$`) so we can identify quickly by static analysis (not running and instecting the actuall code) that we are dealing with a jquery element/s.
+
+```js
+var $elements = $('someSelector'), // in case a global search
+    $element = this.$('some selector'); // in case of backbone
+```
+### Variables: deferred
+
+we recomend using `dfd or promise` as part of a deferred name.
+
+```js
+var requestDfd = fetch('http://some/url'),
+    fetchPromise = new $.Deferred() // in case of jquery promises.
+```
+
+### Variables: objects, strings, integers
+
+Avoid names that are generic and instead, try to make your best effort to find a proper name in order to explain the content of it. If data needs to be changed, as part of a process, try to create methods to produce this changes on demand.
+
+```js
+// bad
+var propsObject = {},
+    trimmedString = 'hello   '.trim(),
+    intPrice = parseInt(5, 10);
+```
+
+### Method: private
+
+we recommend using `_` at the begining of the method.
+
+```js
+// good
+var _getInternalValue = function(){};
+
+// bad
+var getInternalValuePleaseDonot = function(){};
+
+```
+
+In order to undestand what is exposed (the API) for every component that we create/use is important to be able to compartamentalize what is exposed and ready to be used (and to keep and maintain) and what can be changed / discarded.
```

Comment:
**it is** important
**compartmentalize**
maybe change the last `and` to `versus`?